toolchain/check/facet_type.cpp

```diff
@@ -88,7 +84,18 @@ auto InitialFacetTypeImplWitness(
   const auto& facet_type_info =
       context.facet_types().Get(facet_type.facet_type_id);
 
-  if (!is_definition && facet_type_info.rewrite_constraints.empty()) {
+  auto rewrites_into_interface_to_witness = llvm::make_filter_range(
+      facet_type_info.rewrite_constraints,
+      [&](const SemIR::FacetTypeInfo::RewriteConstraint& rewrite) {
+        auto access = context.insts().GetAs<SemIR::ImplWitnessAccess>(
+            GetImplWitnessAccessWithoutSubstitution(context, rewrite.lhs_id));
+        return WitnessQueryMatchesInterface(context, access.witness_id,
+                                            interface_to_witness);
```

Comment:
Comment using words to describe what is going on here? Maybe something like:
```suggestion
      // A rewrite constraint requires `interface_to_witness` to have a witness
      // table if its left-hand side names a member of that interface.
      [&](const SemIR::FacetTypeInfo::RewriteConstraint& rewrite) {
        auto access = context.insts().GetAs<SemIR::ImplWitnessAccess>(
            GetImplWitnessAccessWithoutSubstitution(context, rewrite.lhs_id));
        return WitnessQueryMatchesInterface(context, access.witness_id,
                                            interface_to_witness);
```

Reply:
Added this:
```
  // An iterator over the rewrite_constraints where the LHS of the rewrite names
  // a member of the `interface_to_witness`. This filters out rewrites into
  // other interfaces, as they do not set values in the witness table.
```